backends/cxxrtl/runtime/cxxrtl/cxxrtl.h

```diff
@@ -1769,6 +1769,10 @@ value<BitsY> shr_uu(const value<BitsA> &a, const value<BitsB> &b) {
 template<size_t BitsY, size_t BitsA, size_t BitsB>
 CXXRTL_ALWAYS_INLINE
 value<BitsY> shr_su(const value<BitsA> &a, const value<BitsB> &b) {
+	if (BitsY > BitsA) {
+		value<BitsY> extended_a = a.template scast<BitsY>();
+		return extended_a.shr(b);
+	}
 	return a.shr(b).template scast<BitsY>();
```

Comment:
Would the following one line change not work? It's been a while since I touched that code but `struct sext_cast` already includes the conditional I think.

```suggestion
	return a.template scast<BitsY>().shr(b);
```

Reply:
> `struct sext_cast` already includes the conditional I think

Yes, I originally intended to make this change as well. However, I wasn't sure if BitsY could be smaller than BitsA, so I only handled the case where BitsY is greater than BitsA, while keeping the original functionality of the function intact. If BitsY is guaranteed to be greater than or equal to BitsA, then I think your modification is more elegant and appropriate.

Reply:
I don't think there are restrictions like that on BitsY (see rtlil.cc:1145) but `.template scast()` does handle the case where the result is smaller than `this`: see cxxrtl.h:321 (it uses enable_if to do so).

Reply:
You are right, I didn't notice the role of `cxxrtl.h:321`. Your fix is more appropriate, thank you for the correction :).